src/background/services/storage.ts

```diff
@@ -253,9 +245,18 @@ const MIGRATIONS: Record<Storage['version'], Migration> = {
     }
 
     if (data.hasHostPermissions === false) {
-      data.state = 'missing_host_permissions' satisfies Storage['state']
+      data.state = 'missing_host_permissions'
     }
 
     return [data, deleteKeys]
+  },
+  3: (data) => {
+    const state = data.state
+    if (!state) {
+      data.state = null
```

Comment:
Can't we have it as an empty object?

Reply:
Yeah, I think that `null` optimization is unnecessary.